packages/helpers/src/extractChildren.js

```diff
@@ -3,8 +3,13 @@ import React from "react";
 export default function extractChildren(children, types) {
   const _children = [];
   const components = {};
+
   if (Array.isArray(types)) {
-    React.Children.toArray(children).forEach(child => {
+    for (const child of React.Children.toArray(children)) {
```

Comment:
What is the benefit of `for (child of ...` vs `.forEach(...`?

Reply:
I had to return the final results early if there's React.Fragment

Reply:
[this transpiled code tho](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=GYewTgBAFAxiB2BnALhAHhEwIE8CUEA3gL5A&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ces2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.12.9&externalPlugins=)
![image](https://user-images.githubusercontent.com/93752/100673523-73cd5400-3318-11eb-933a-907c4b6642b5.png)

Reply:
wao, that's a lot of code for just the `const x of y` didn't know that, any way to just ship that for browsers that don't support it natively? @alexzherdev ?

Reply:
seem like all that code is just for IE11 correct? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
I can use `some` or `every` then!

Reply:
> seem like all that code is just for IE11 correct? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

pretty much yeah

> wao, that's a lot of code for just the `const x of y` didn't know that, any way to just ship that for browsers that don't support it natively? @alexzherdev ?

paprika is transpiled and some level of browser support has to be set for that transpilation. I'm not aware of an easy way to publish two sets of artifacts (e.g. modern vs IE) that could be seamlessly consumed and further used in an app. It's possible to transpile paprika just enough to be evergreen-browser-compatible, and then configure apps to produce two bundles (where IE bundle would transpile paprika's code further). But that'd be a pretty significant change